src/Gaffer/ValuePlug.cpp

```diff
@@ -624,8 +626,24 @@ class ValuePlug::ComputeProcess : public Process
 			const ThreadState &threadState = ThreadState::current();
 			const Context *currentContext = threadState.context();
 
-			const ComputeNode *computeNode = IECore::runTimeCast<const ComputeNode>( p->node() );
-			const ComputeProcessKey processKey( p, plug, computeNode, computeNode ? computeNode->computeCachePolicy( p ) : CachePolicy::Uncached, precomputedHash );
+			CachePolicy cachePolicy = CachePolicy::Uncached;
+			if( computeNode )
+			{
+				if( !p->getInput() )
+				{
+					cachePolicy = computeNode->computeCachePolicy( p );
+				}
+				else
+				{
+					// I'm not sure we even need to cache plugs that are driven by an input,
+					// and only need a type conversion, but currently, we have a special hash
+					// for these in HashProcess::hash(), so we'll continuing using that hash
+					// in Legacy mode, matching previous behaviour.
```

Comment:
I'm not sure either, but `StringPlug::setFrom( StringVectorDataPlug )` can perform an arbitrarily large amount of work depending on the input value, so we'd probably want to measure the effects across a variety of real-world graphs, and maybe consider possible future type conversions.

In any case, we're not caching _because_ we have a special hash, so I'd suggest a comment like this might be more helpful :

```
/// Type conversion will be implemented by `setFrom()`.
/// \todo Determine if caching is actually worthwhile for this.
```

Reply:
Tiny note here just to make sure I'm not missing something: you've used triple slashes in you comment. I've taken it verbatim, but used double slashes instead, to match the other comments in the file. My understanding is that triple slashes are only for comments on function signatures that we want to be picked up by documentation generators, unless I've missed something.

Reply:
> My understanding is that triple slashes are only for comments on function signatures that we want to be picked up by documentation generators, unless I've missed something.

Yes, but with the caveat that they're also needed to pick up `\todo` entries by the same documentation generators. But we're not actually using said documentation generators currently, and if we did, we'd be unlikely to include the todos in the API docs, so I don't think it matters.